packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -291,6 +294,10 @@ async function nodeImport(
         ? { ...resolveOptions, tryEsmOnly: true }
         : resolveOptions
     )
+    // These cannot be imported directly
+    if (url.match(/\.(css|less|sass|scss|styl|stylus|pcss|postcss|json)$/)) {
```

Comment:
I agree but I don't think `marko`, `svelte`, or `astro` files can be directly imported by Node, can they? If we use `isJSRequest` `nodeImport` will try a direct import. It would also miss `wasm` files (I don't know whether it's desirable or not). I couldn't find a good regex for this anywhere.

How about actually trying to import directly and catching `ERR_UNKNOWN_FILE_EXTENSION` to try again with Vite's loaders? This way we could support custom Node loaders.